fix: key upload handling for SAML and OpenID

Foorcee opened the pull request:
Improves the file upload for certificates and key stores. Adds form validation and provides more detailed error messages.

Closes #38049

edewit reviewed (changes requested):
We don't need to change the public interface just to show a different message, you could have just try catch the `asString` method instead

Foorcee committed:
fix: key upload handling for SAML and OpenID
Improves the file upload for certificates and key stores. Adds form validation and provides more detailed error messages.

Closes #38049

Signed-off-by: Marco Neuhaus <mneuhaus44@gmail.com>

ahus1 reviewed (approved):
Approving based on @edewit's review. I also did a manual local test.

ahus1 commented:
@Foorcee - this PR will be part of Keycloak 26.3.0 once it is released. If you want this change to be part of a 26.2.x patch release, you can help us out by creating a backport with a pull request against the `release/26.2` branch.

ahus1 commented:
@Foorcee - I noticed that when uploading a new certificate, the currently shown certificate did not change. The new certificate was only showing after I refreshed the browser. 

If you would want to tackle this glitch, please create a GitHub bug issue and provide a pull request. Feel free to ping me to get it reviewed. Thanks!
